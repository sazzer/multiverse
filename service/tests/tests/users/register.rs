use crate::{service::TestService};
use insta::{assert_debug_snapshot, assert_json_snapshot};
use serde_json::json;

#[actix_rt::test]
async fn integration_test_register_no_fields() {
    let service = TestService::new().await;

    let response = service.request(actix_web::test::TestRequest::post().uri("/users").set_json(&json!({})).to_request()).await;

    assert_debug_snapshot!(response.headers(), @r###""HTTP/1.1 422 Unprocessable Entity\ncontent-type: application/problem+json""###);
    assert_json_snapshot!(response.to_json().unwrap(), @r###"
    {
      "fields": {
        "email_address": {
          "title": "The required field was missing",
          "type": "tag:multiverse,2020:problems/validation_error/missing"
        },
        "password": {
          "title": "The required field was missing",
          "type": "tag:multiverse,2020:problems/validation_error/missing"
        },
        "username": {
          "title": "The required field was missing",
          "type": "tag:multiverse,2020:problems/validation_error/missing"
        }
      },
      "status": 422,
      "title": "A validation error occurred",
      "type": "tag:multiverse,2020:problems/validation_error"
    }
    "###);
}
