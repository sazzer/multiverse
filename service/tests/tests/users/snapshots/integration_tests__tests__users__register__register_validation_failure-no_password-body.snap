---
source: tests/tests/users/register.rs
expression: response.to_json().unwrap()
---
{
  "fields": {
    "password": {
      "title": "The required field was missing",
      "type": "tag:multiverse,2020:problems/validation_error/missing"
    }
  },
  "status": 422,
  "title": "A validation error occurred",
  "type": "tag:multiverse,2020:problems/validation_error"
}
